src/mmw/apps/bigcz/clients/cuahsi/details.py

```diff
@@ -4,10 +4,19 @@
 from __future__ import division
 
 from datetime import date, timedelta
+from timeout_decorator import timeout
 
-from rest_framework.exceptions import ValidationError
+from rest_framework import status
+from rest_framework.exceptions import APIException, ValidationError
 
 DATE_FORMAT = '%m/%d/%Y'
+WOFPY_TIMEOUT = 5
+
+
+class WOFpyTimeout(APIException):
+    status_code = status.HTTP_504_GATEWAY_TIMEOUT
+    default_detail = \
+        'WOFpy request did not finish in {} seconds'.format(WOFPY_TIMEOUT)
```

Comment:
Is it useful to say `WOFpy` in the error message for the typical user? It could potentially stay as is without `WOFpy`.

Reply:
Admittedly this was a little defensive, in the sense we want to differentiate between an MMW timeout (our fault) and a WOFpy timeout (not our fault). But if it feels too unnecessarily detailed we can simplify it.

Reply:
I'm just not sure that anyone will know what it is. Maybe something like `Details request did not finish in {} seconds`? Not a dealbreaker though.